### ISSUE TITLE
The task fails on directory creation

### ISSUE THREAD

**Issue by lukehler** [uhop/grunt-slice-front#1]
While trying to dynamically map all files in a source directory to a clean destination directory, I was found that the task would fail, since it tries to open any files in subdirectories without first creating the those subdirectories.

Here's a quick test configuration you could add to the Gruntfile, just to illustrate. You'll see the task fail as it tries to write to tests/dynamic.

```js
// Read all files from a source directory (tests/) and write the results to a new directory (tests/dynamic/)
dynamic: {
  options: {
    templateFile: "tests/trivial/trivial.html.jst",
    templateParams: {
      title: "Dynamic demo"
    }
  },
  // dynamically mapping files, so it's not just reading demo.md, but also subpage/demo.md, as well.
  files: [
    {
      expand: true,
      cwd: "tests/",
      src:  [
        "*.md",
        "**/*.md"
      ],
      dest: "tests/dynamic/",
      ext: ".html"
    }
  ]
}
```

The same issue becomes really relevant if you have a task that takes the following:
- src/index.md
- src/about/index.md

... and tries to write the results to a clean destination folder:
- dest/index.html
- dest/about/index.html

Using `grunt.file`  instead of `fs` resolves this pretty tidily without introducing any new dependencies. I'll get a pull request underway, if you're interested.

**Comment by uhop** [comment]
Thank you for the report. It is clearly a bug, and `grunt.file` solution sounds optimal. Looking forward to see a pull request.

**Comment by uhop** [comment]
The problem is resolved by pull request #2 --- thx @lukehler!
